### ISSUE TITLE
Bundle size very large

### ISSUE THREAD

**Issue by simonlc** [giovannibenussi/use-persisted-state-hook#3]
Hi, I was checking out this library and it sounds very great. Though I have one concern after viewing it on bundlephobia: https://bundlephobia.com/package/use-persisted-state-hook@1.0.1

The problem seems to be the usage of object-hash. I'm not entirely familiar with how it's being used but I was wondering if a lighter alternative could be used?

Thanks.

**Comment by giovannibenussi** [comment]
Hi, I noticed this at the beginning too but it wasn’t the big detail at first because I wasn’t sure if people were going to use this library, I’m glad that you liked it!

I’m using object-hash to detect changes in the initial state, so I know that I need to reset the persisted state if the initial state changes. For example, let say that you start with this state in the first version of your app:

```
useLocalStorageState('visitCount',  0)
```

After months using it, you decide to change it to a hash to be able to have more data within  the same key:

```
useLocalStorageState('visitCount',  { total: 0, today: 0 })
```

A user with old data from the first version may have issues if you perform actions assuming that data is a hash instead of a number (access a key for example). The safest option is to reset the persisted state if the initial state changes, and the only way into do that is to persist an identifier of the initial value using object-hash.

I’ll take a look at this during the week to see if I can work on this to reduce the bundle size. I’m pretty sure there’s a way that work for us without requiring an external library!

**Comment by simonlc** [comment]
That's great to hear, thank you!

Really cool to see a library that handles cases like this, not just a simple locationStorage adapter.

**Comment by samjbobb** [comment]
In this use case, I think something like this would be a suitable replacement for the `hash` function from `object-hash`:

`(value) => md5(JSON.stringify(value))`

I think object-hash is providing some features that aren’t needed here:

* ensuring consistent hashes for objects across environments with slightly different behavior of JSON.stringify (like white space handling). This isn’t needed for this library since the hash is only compared within a single browser.
* hashing for non-enumerable properties. Also doesn’t matter for this library because the state is being JSON.stringify’d for localStorage anyway. 

In short, I think calculating a hash of the JSON string of the object would be a quick and easy way to remove the object-hash dependency. 

What do you think?

**Comment by giovannibenussi** [comment]
Sorry for the delay on this! You are right @samjbobb, I tested your proposed solution and it effectively works. I [created a PR](https://github.com/giovannibenussi/use-persisted-state-hook/pull/4) to apply this change. I plan to merge it this week in case you want to take a look at it!

**Comment by giovannibenussi** [comment]
I just merged the changes and released version 1.0.2. Bundle size went from [10.6 kb](https://bundlephobia.com/package/use-persisted-state-hook@1.0.1) to [1.9 kb](https://bundlephobia.com/package/use-persisted-state-hook@1.0.2) when compressed.

Thanks a lot for the help of both of you on this!

**Comment by samjbobb** [comment]
Looks good @giovannibenussi. Thanks for the fix!
